tools/stix-to-ecs/stix_to_ecs.py

```diff
@@ -1,588 +1,659 @@
+            case patterns.ParentheticalExpression:
+                self.__visit(o.expression)
+            case patterns.EqualityComparisonExpression:
+                self.__visit_equality_comparison_expression(o)
+            case _:
+                raise NotImplemented(ot)
+        return self
+
+    def __visit_equality_comparison_expression(self, o) -> STIXToECSPatternParser:
+        lhs = o.lhs
+        rhs = o.rhs
+
+        if not (t := type(lhs)) == patterns.ObjectPath:
+            raise NotImplemented(t)
+
+        if not (t := type(rhs)) == patterns.StringConstant:
+            raise NotImplemented(t)
+
+        self.__loot.append(
+            (
+                convert_stix_word_to_ecs_word(lhs.object_type_name),
+                convert_stix_word_to_ecs_word(lhs.property_path[0].property_name),
+                ":".join(
+                    [
+                        convert_stix_word_to_ecs_word(x.property_name)
+                        for x in lhs.property_path[1:]
+                    ]
+                    + [rhs.value]
+                ),
+            )
+        )
+
+        return self
+
+    @property
+    def data(self) -> dict:
+        """
+        Get ECS data.
+        """
+
+        return copy.deepcopy(self.__data)
+
+    @property
+    def description(self) -> str:
+        """
+        Get ECS description.
+        """
+
+        if type(self.data) is dict and (hash := self.data.get("hash", None)):
+            # ctf -> SHA256 may not be always available, in this case do we want to crash or take an other algo?
+            observable = hash.get("sha256", None) or hash.get("sha1", None) or hash.get("md5", None)
+            if not observable:
+                raise RuntimeError("Missing SHA256, SHA1, or MD5 observable from set of hashes.")
+        else:
+            observable = self.data
+
+        return f"Simple indicator of observable {{{observable}}}"
+
+    @property
+    def field_name(self) -> str:
+        """
+        Get ECS field name.
+        """
+
+        return self.__field_name
+
+    @property
+    def type(self) -> str:
+        """
+        Get ECS type.
+        """
+
+        return self.__type
+
+
+def check_arguments(args: argparse.Namespace) -> bool:
+    """
+    The function check if the provided arguments are correctly used.
+    :param args: The parsed arguments.
+    :return: True if the arguments are corrects, False otherwise.
+    """
+
+    if args.recursive and not args.input.is_dir():
+        print("argument -r only allowed if input is a directory")
+        return False
+
+    if args.elastic:
+        if args.configuration:
+            return True
+
+        for x in ["cloud_id", "index"]:
+            if not getattr(args, x):
+                print(
+                    f"argument --{x} is required to connect to the Elastic cluster (-e, --elastic) unless configuration file is provided (-c, --configuration)"
+                )
+                return False

```

Comment:
Logic need to be extended to take into account the new `--username` and `--password` parameter in case a `url` is provided, we should also check that we can't provide and a `cloud_id` and a `url`